test/e2e/storage/host_path_type.go

```diff
@@ -0,0 +1,342 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = utils.SIGDescribe("HostPathType", func() {
+	f := framework.NewDefaultFramework("host-path-type")
+
+	var (
+		ns           string
+		hostBaseDir  string
+		mountBaseDir string
+		basePod      *v1.Pod
+
+		hostPathUnset             = v1.HostPathUnset
+		hostPathDirectoryOrCreate = v1.HostPathDirectoryOrCreate
+		hostPathDirectory         = v1.HostPathDirectory
+		hostPathFileOrCreate      = v1.HostPathFileOrCreate
+		hostPathFile              = v1.HostPathFile
+		hostPathSocket            = v1.HostPathSocket
+		hostPathCharDev           = v1.HostPathCharDev
+		hostPathBlockDev          = v1.HostPathBlockDev
+	)
+
+	BeforeEach(func() {
+		ns = f.Namespace.Name
+
+		By("Create a pod for further testing")
+		hostDir := "/tmp/hostpath-test"
+		mountDir := "/mnt/test"
```

Comment:
It seems like all the tests cases use the same mountDir.  Does it need to be an argument?